### ISSUE TITLE
Filter for Talks page

### ISSUE THREAD

**Issue by marcoroth** [adrienpoly/rubyvideo#507]
Either we should have a `Past Talks` and `Upcoming Talks` page or we should have a filter on the `/talks` page, since the first page only show scheduled talks now 😅 

![CleanShot 2024-12-10 at 01 37 50](https://github.com/user-attachments/assets/94173113-17d6-47c5-be29-ac8d53c1c7f7)

**Comment by adrienpoly** [comment]
yes I noticed that and I found it a bit annoying. Technically the controller already [support filters](https://github.com/adrienpoly/rubyvideo/pull/433) it would be fairly easy to add the video_provider. The missing piece is the UI for all those filters. I have started to look at a few options from other site. This is clearly something that I would like to push forward 

For this particular case we could have a video_provider (name?) filter with a default value that filters out the upcoming talks

**Comment by justinallenmarsh** [comment]
@marcoroth @adrienpoly I came up with a filter for this. There is a scope added by #530 which already filters upcoming out.

https://share.cleanshot.com/4flshVxH

Do we want to go with a solution that allows users to filter by what they want specifically? maybe "Published" can be default.

**Comment by adrienpoly** [comment]
It looks great thanks. I would go with the published by default.
The components takes quite a bit of space but right now it is not a big issue. Further down the road we might want to add more filters and we might need more space. Maybe we would have to move to a select at some point ? But let's keep it like that for now.

How does it looks on mobile?

**Comment by justinallenmarsh** [comment]
Here is a link of how it works on mobile.

**Mobile**
https://share.cleanshot.com/kfrgc1sr

**Desktop**
https://share.cleanshot.com/BZctgK6J

I also think it takes up a lot of space, but it fits at the moment like this. I think if we add more filters and select then we can create a modal window.

I changed the wording to align a little bit more with the data model and also used "All" instead of "All Talks" to conserve space.

You can find the PR here. #927
